common/hyperscaler/rules/parsing_result.go

```diff
@@ -0,0 +1,47 @@
+package rules
+
+import "github.com/google/uuid"
+
+type ParsingResult struct {
+	ID uuid.UUID
+
+	OriginalRule string
+
+	Rule *Rule
+	// array with errors that occurred during parsing of rule entry
+	ParsingErrors []error
+
+	// array with errors that occurred after successful rule parsing
+	ProcessingErrors []error
+}
+
+func NewParsingResult(originalRule string, rule *Rule) *ParsingResult {
+	return &ParsingResult{
+		ID:               uuid.New(),
+		OriginalRule:     originalRule,
+		ParsingErrors:    make([]error, 0),
+		ProcessingErrors: make([]error, 0),
+		Rule:             rule,
+	}
+}
+
+func (r *ParsingResult) HasParsingErrors() bool {
+	return len(r.ParsingErrors) > 0
+}
+
+func (r *ParsingResult) HasProcessingErrors() bool {
+	return len(r.ProcessingErrors) > 0
+}
+
+func (r *ParsingResult) HasErrors() bool {
+	return r.HasParsingErrors() || r.HasProcessingErrors()
+}
+
+// TODO: Change name to Uniqnuess errors or something like that
+func (r *ParsingResult) AddProcessingError(err error) {
```

Comment:
This extraction seems excessive.
Could we just inline appends?

Reply:
It seems good for me.

Reply:
Any rationale? Justification?
Is this somehow helping us at cost of being slightly harder to read?

Reply:
Encapsulation. Feel free to change implementation however you want.